cmd/argocd/commands/app.go

```diff
@@ -1164,25 +1164,25 @@ func NewApplicationDiffCommand(clientOpts *argocdclient.ClientOptions) *cobra.Co
 			errors.CheckError(err)
 			diffOption := &DifferenceOption{}
 			if app.Spec.HasMultipleSources() && len(revisions) > 0 && len(sourcePositions) > 0 {
-
-				revisionSourceMappings := make(map[int64]string, 0)
-				for i, pos := range sourcePositions {
-					if pos <= 0 {
-						errors.CheckError(fmt.Errorf("source-position cannot be less than or equal to 0. Counting starts at 1."))
+				numOfSources := int64(len(app.Spec.GetSources()))
+				for _, pos := range sourcePositions {
+					if pos <= 0 && pos > numOfSources {
+						log.Fatal("source-position cannot be less than or equal to 0. Counting starts at 1.")
 					}
```

Comment:
Should this be `||` instead of `&&`? If so, the error message should probably be updated too.